packages/asc-ui/src/components/Modal/Modal.tsx

```diff
@@ -36,28 +36,33 @@ const Modal: FunctionComponent<Props> = ({
 }) => {
   const ref = useRef<HTMLDivElement>(null)
   const { keyDown } = useTrappedFocus(ref)
+  const previouslyFocusedElement = document.activeElement
 
   useEffect(() => {
-    let renderedTimer = 0
-    const { current: node } = ref
-    if (node) {
-      clearTimeout(renderedTimer)
-      renderedTimer = setTimeout(() => {
-        node.focus()
-      })
-    }
-    return () => {
-      clearTimeout(renderedTimer)
+    if (open && ref.current) {
+      const node = ref.current
+      const firstFocusableEl = node.querySelector(FocusableElements.join(', '))
+
+      if (firstFocusableEl instanceof HTMLElement) {
+        firstFocusableEl.focus()
+      }
     }
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  }, [open])
 
   const handleClose = () => {
     if (onClose) {
+      if (
+        previouslyFocusedElement &&
+        previouslyFocusedElement instanceof HTMLElement
+      ) {
```

Comment:
Since `instanceof` will also check for `undefined` and other empty values there is no need to do an extra check.
```suggestion
 if (previouslyFocusedElement instanceof HTMLElement) {
```